docs/cn/built_in_function_develop_guide.md

```diff
@@ -0,0 +1,709 @@
+
+- ```c++
+  # hybridse/src/udf/udf.h
+  namespace hybridse {
+    namespace udf {
+      namespace v1 {
+        // declare built-in function
+      } // namespace v1
+    } // namespace udf
+  } // namespace hybridse
+  ```
+
+- ```c++
+  # hybridse/src/udf/udf.cc
+  namespace hybridse {
+    namespace udf {
+      namespace v1 {
+        // implement built-in function
+      } // namespace v1
+    } // namespace udf
+  } // namespace hybridse
+  ```
+
+#### 2.1.2 C++函数名规范
+
+- C++内置函数名统一使用[snake_case](https://en.wikipedia.org/wiki/Snake_case)风格
+- 要求函数名能清晰表达函数功能
+- **不要求**与对外暴露的SQL函数名**完全一致**。因为C++函数与SQL函数名是通过函数注册操作关联在一起
+
+#### 2.1.3 C++类型与SQL类型对应关系
+
+内置C++函数的参数类型限定为：BOOL类型，数值类型，时间戳日期类型和字符串类型。C++类型SQL类型对应关系如下：
+
+| SQL类型   | C/C++ 类型         |
+| :-------- | :----------------- |
+| BOOL      | `bool`             |
+| SMALLINT  | int16_t            |
+| INT       | `int32_t`          |
+| BIGINT    | `int64_t`          |
+| FLOAT     | `float`            |
+| DOUBLE    | `double`           |
+| STRING    | `codec::StringRef` |
+| TIMESTAMP | `codec::Timestamp` |
+| DATE      | `codec::Date`      |
+
+#### 2.1.4 函数参数和返回值
+
+- C++函数与SQL函数参数相对位置相同
+
+- C++函数与SQL函数与参数类型类型一一对应。详情参见[2.1.3 C++类型与SQL类型对应关系](#2.1.3-C++类型与SQL类型对应关系)
+
+- 函数返回值要考虑以下几种情况：
+
+  - 当SQL函数返回值为布尔或数值类型（如**BOOL**, **SMALLINT**, **INT**, **BIGINT**, **FLOAT**, **DOUBLE**）时，一般地，C++的函数返回值为与之对应的C++类型（`bool`, `int16_t`, `int32_t`, `int64_t`, `float`, `double`)。
+
+    - ```c++
+      // SQL: DOUBLE FUNC_DOUBLE(INT)
+      double func_return_double(int); 
+      ```
+
+  - 当SQL函数返回值为**STRING**, **TIMESTAMP**或**DATE**时，要求C++函数通过参数来输出结果。这意味着，C++的**输入参数**后额外有一个指针类型的**输出参数**（`codec::StringRef*`, `codec::Timestamp*`或 `codec::Date*`)用来存放和返回结果值
+
+    - ```c++
+      // SQL: STRING FUNC_STR(INT)
+      void func_output_str(int32_t, codec::StringRef*); 
+      ```
+
+  - 当SQL函数的返回值可能为空(**Nullable**)时，额外要有一个`bool*`类型的**输出参数**来存放结果为空与否
+
+    - ```c++
+      // SQL: Nullable<DATE> FUNC_NULLABLE_DATE(BIGINT)
+      void func_output_nullable_date(int64_t, codec::Date*, bool*); 
+      ```
+    
+  - 注意， SQL 函数返回值将对内置函数的实现和注册方式产生较大的影响，我们将在后续分别讨论，详情参见[3. SQL函数开发模版](#3.-SQL函数开发模版)。
+
+### 2.2. 注册和配置接口
+
+#### 2.2.1 注册位置
+
+OpenMLDB的 `DefaultUdfLibrary` 负责存放和管理内置的全局SQL函数。默认函数库`DefaultUdfLibrary` 声明在文件[hybridse/src/udf/default_udf_library.h](https://github.com/4paradigm/OpenMLDB/blob/main/hybridse/src/udf/default_udf_library.h)文件中，实现在[hybridse/src/udf/default_udf_library.cc](https://github.com/4paradigm/OpenMLDB/blob/main/hybridse/src/udf/default_udf_library.cc)文件中。
+
+开发者先在[hybridse/src/udf/default_udf_library.cc](https://github.com/4paradigm/OpenMLDB/blob/main/hybridse/src/udf/default_udf_library.cc)中将C++函数注册到默认函数库 `DefaultUdfLibrary` 中，用户才能通过SQL语句访问这个函数。为了更好的管理函数注册，`DefaultUdfLibrary`提供几类`DefaultUdfLibrary::InitXXXXUdf()` 接口来管理不同功能的函数配置注册:
+
+- 数学函数：在`DefaultUdfLibrary::IniMathUdf()`中注册
```

Comment:
I have renamed this function to `InitMathUdf` in https://github.com/4paradigm/OpenMLDB/pull/791

Reply:
Resolved